misc/docs-linter/main.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"golang.org/x/sync/errgroup"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+)
+
+var (
+	ErrEmptyPath = errors.New("you need to pass in a path to scan")
+)
+
+type cfg struct {
+	docsPath string
+}
+
+func main() {
+	cfg := &cfg{}
+
+	cmd := commands.NewCommand(
+		commands.Metadata{
+			Name:       "docs-linter",
+			ShortUsage: "docs-linter [flags]",
+			ShortHelp:  "Finds broken 404 links in the .md files in the given folder & subfolders",
+		},
+		cfg,
+		func(ctx context.Context, args []string) error {
+			return execLint(cfg, ctx)
+		})
+
+	cmd.Execute(context.Background(), os.Args[1:])
+}
+
+func (c *cfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.StringVar(
+		&c.docsPath,
+		"path",
+		"./",
+		"path to dir to walk for .md files",
+	)
+}
+
+func execLint(cfg *cfg, ctx context.Context) error {
```

Comment:
If this is supposed to be a linting suite for the documentation, then the main run loop shouldn't need to run a single linter, but be able to run multiple ones

Please create some kind of pipeline system 🙏

Reply:
The idea with this was to only have a link 404 checker. [Reference this comment](https://github.com/gnolang/gno/pull/2125#discussion_r1624681279)

Reply:
It is intended to become a linting suite once we determine a second linting feature to implement.

It's a `main` package that won't be imported. I'm fine with adding the pipeline system later.